plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/TestPrometheusIntegrationStatus.java

```diff
@@ -100,8 +102,8 @@ public void testConfirmMetricAvailableAndCheckUp()
     @Test
     public void testPushDown()
     {
-        // default interval on the `up` metric that Prometheus records on itself is 15 seconds, so this should only yield one row
+        // default interval on the `up` metric that Prometheus records on itself is about 15 seconds, so this should only yield one or two row
         MaterializedResult results = computeActual("SELECT * FROM prometheus.default.up WHERE timestamp > (NOW() - INTERVAL '15' SECOND)");
```

Comment:
Can we instead create a timestamp literal in Java side and then send that as the predicate.
In the MaterializedResult then we can assert that all rows have a timestamp > the predicate (instead of counting number of rows which is inherently flaky).

Reply:
Also if we want to test pushdown happenned we should assert on the plan. Since even without pushdown the engine will filter results and we'll get filtered rows.

It's pre-existing issue though.

Reply:
Let me punt to #10784

Reply:
Do we have some stress test results somewhere? LGTM.

Reply:
Tested locally because the failure ratio was very low (1 / 500 attempts) in my laptop. Verified 2000 passes with this fix.